diff.py

```diff
@@ -2847,184 +2847,192 @@ def process(dump: str, config: Config) -> List[Line]:
+                if re.search(arch.re_reloc, reloc_row):
+                    original, reloc_symbol = processor.process_reloc(
+                        reloc_row, original
+                    )
+                    if reloc_symbol is not None:
+                        symbol = reloc_symbol
+                else:
+                    break
+                i += 1
+
+            is_text_relative_j = False
+            if (
+                arch.name in MIPS_ARCH_NAMES
+                and mnemonic == "j"
+                and symbol is not None
+                and symbol.startswith(".text")
+            ):
+                symbol = None
+                original = row
+                is_text_relative_j = True
+
+            normalized_original = processor.normalize(mnemonic, original)
+
+            scorable_line = normalized_original
+            if not config.score_stack_differences:
+                scorable_line = re.sub(arch.re_sprel, "addr(sp)", scorable_line)
+
+            row = re.sub(arch.re_reg, "<reg>", row)
+            row = re.sub(arch.re_sprel, "addr(sp)", row)
+            if mnemonic in arch.instructions_with_address_immediates:
+                row = row.strip()
+                row, _ = split_off_address(row)
+                row += "<imm>"
             else:
-                # Then, we try to match the global deref in a separate regex.
-                x86_longjmp = re.search(r"\*(.*)", args)
+                row = normalize_imms(row, arch)
+
+            branch_target = None
+            if (
+                mnemonic in arch.branch_instructions or is_text_relative_j
+            ) and symbol is None:
+                # Here, we try to match a wide variety of addressing mode:
+                # - Global deref with offset: *0x1234(%eax)
+                # - Global deref: *0x1234
+                # - Register deref: *(%eax)
+                #
+                # We first have a single regex to match register deref and global
+                # deref with offset
+                x86_longjmp = re.search(r"\*(.*)\(", args)
                 if x86_longjmp:
                     capture = x86_longjmp.group(1)
                     if capture != "" and capture.isnumeric():
                         branch_target = int(capture, 16)
                 else:
-                    branch_target = int(args.split(",")[-1], 16)
+                    # Then, we try to match the global deref in a separate regex.
+                    x86_longjmp = re.search(r"\*(.*)", args)
+                    if x86_longjmp:
+                        capture = x86_longjmp.group(1)
+                        if capture != "" and capture.isnumeric():
+                            branch_target = int(capture, 16)
+                    else:
+                        branch_target = int(args.split(",")[-1], 16)
+        except Exception as e:
+            # If we fail to parse the line, at least emit something rather than crashing.
+            original = str(e)
```

Comment:
Hm, we're constructing a Line with a whole bunch of properties after this, but they aren't set if an exception is thrown. I guess we should assign empty values to them before the start of the try block?

Reply:
or better, do `output.append(Line(...)); continue` like we do in some other cases within the try, and just ignore all of the properties normally passed when constructing the Line

Reply:
Sure - I also added the num_instr +=1 and the source_lines = [] that also happens in the normal case